app/actions/app_apply_manifest.rb

```diff
@@ -140,30 +140,35 @@ def create_service_bindings(manifest_service_bindings_message, app)
       manifest_service_bindings_message.manifest_service_bindings.each do |manifest_service_binding|
         service_instance = app.space.find_visible_service_instance_by_name(manifest_service_binding.name)
         service_instance_not_found!(manifest_service_binding.name) unless service_instance
-        binding_being_deleted!(service_instance, app)
-        next if binding_exists?(service_instance, app)
-
-        begin
-          binding_message = create_binding_message(service_instance.guid, app.guid, manifest_service_binding)
-          action = V3::ServiceCredentialBindingAppCreate.new(@user_audit_info, binding_message.audit_hash, manifest_triggered: true)
-          binding = action.precursor(
-            service_instance,
-            app: app,
-            volume_mount_services_enabled: volume_services_enabled?,
-            message: binding_message)
+
+        binding = ServiceBinding.first(service_instance: service_instance, app: app)
+        unless binding&.create_succeeded?
+
+          raise_invalid_binding_error!(service_instance, binding) unless !binding || binding.create_failed?
```

Comment:
I find this condition "unless not .. or" hard to understand. By changing this to two dedicated methods and conditions, it becomes more readable:
```
binding_operation_in_progress!(service_instance, binding.last_operation.type) if binding&.operation_in_progress?
binding_deletion_failed!(service_instance) if binding&.delete_failed?
```